components/sync15/src/request.rs

```diff
@@ -24,7 +24,10 @@ struct LimitTracker {
     cur_records: usize,
 }
 
+// The implementation for the limit, allowing for a new limit,
+// to reset the limit, and keep track of records count in the queue
 impl LimitTracker {
+    // creates a new limit
```

Comment:
Turn all of these into [doc comments](https://doc.rust-lang.org/rust-by-example/meta/doc.html).